pkgs/development/compilers/flutter/patches/flutter3/set-flutter-gradle-cache.patch

```diff
@@ -0,0 +1,45 @@
+Patch the gradle calls within flutter android code to explicitly 
+set --project-cache-dir from FLUTTER_GRADLE_PROJECT_CACHE
+
+otherwise gradle build will attempt to write to the sdk directory.
+
+diff --git a/packages/flutter_tools/lib/src/android/gradle.dart b/packages/flutter_tools/lib/src/android/gradle.dart
+index 6d1e3e7345..8ef5c630c8 100644
+--- a/packages/flutter_tools/lib/src/android/gradle.dart
++++ b/packages/flutter_tools/lib/src/android/gradle.dart
+@@ -181,9 +181,11 @@ class AndroidGradleBuilder implements AndroidBuilder {
+        _usage = usage,
+        _gradleUtils = gradleUtils,
+        _androidStudio = androidStudio,
++       _flutterProjectCacheDir = platform.environment["FLUTTER_GRADLE_PROJECT_CACHE"] ?? "${platform.environment["HOME"]}/.gradle_android_cache",
```

Comment:
I'm not too sold on this default path. Shouldn't we be using an [XDG base directory](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) like XDG_CACHE_HOME?